### DISCUSSION TITLE
Notifikasi suara di Android

### DISCUSSION THREAD

**Discussion by laksa19** [aldinokemal/go-whatsapp-web-multidevice#145] [General]
Ketika login di API ini, WhatsApp di Android tidak mengeluarkan notifikasi suara, 
Setelah logout notifikasi suara kembali normal.

**Comment by aldinokemal** [comment]
Ini bukan bug notifikasi WhatsApp-nya, tapi efek dari presence. Waktu laporan ini dibuat (Mei 2024) aplikasi selalu mengirim `PresenceAvailable` begitu terhubung, lihat `src/pkg/whatsapp/whatsapp.go:193` dan `:217` pada commit `7292058`. Kalau salah satu linked device berstatus online, WhatsApp menganggap kamu sedang aktif di perangkat lain, jadi notifikasi (termasuk suaranya) di HP dibisukan. Setelah logout tidak ada device online lagi, makanya notifikasi kembali normal.

Sejak `v8.3.0` perilaku ini sudah bisa diatur dan defaultnya tidak online lagi:

```
WHATSAPP_PRESENCE_ON_CONNECT=unavailable
```

atau pakai flag `--presence-on-connect=unavailable`. Pilihannya:

- `available`: online, notifikasi di HP ikut senyap
- `unavailable`: pushname tetap terdaftar tanpa online (default sekarang, notifikasi HP tetap jalan)
- `none`: tidak mengirim presence sama sekali (pushname tidak terdaftar, kontak bisa melihat "-" sebagai nama)

Kodenya di `src/infrastructure/whatsapp/event_handler.go:119-141`, nilai default di `src/config/settings.go:63`.

Satu catatan tambahan: sejak `v8.7.0` ada presence pulse yang menandai device sebagai online selama 5 menit setiap 24 jam (`src/infrastructure/whatsapp/presence_pulse.go:170-191`, default `true` di `src/config/settings.go:64`). Selama jendela 5 menit itu notifikasi HP ikut senyap. Kalau mengganggu, matikan dengan:

```
WHATSAPP_PRESENCE_PULSE_ENABLED=false
```

Dan pastikan `WHATSAPP_AUTO_MARK_READ` dibiarkan `false` (default di `src/config/settings.go:45`), karena auto mark read juga bikin notifikasi di HP hilang.
